CIPs/cip-137.md

```diff
@@ -18,187 +18,223 @@ An API for reading and writing events on Ceramic event streams.
 
-**Params:**
+#### Example:
 
-- *ceramicUrl* <string> - the Ceramic URL to subscribe to
-- *options* <object> - options for the subscription
-    - *ctrlRangeStart* <string> - hex encoded string describing the start of the range of controllers to sync (defaults to `0x0000000000000000`)
-    - *ctrlRangeEnd* <string> - hex encoded string describing the end of the range of controllers to sync (defaults to `0xffffffffffffffff`)
+```
+/ceramic/subscribe/*?model=kjzl6hvfrbw6c82mkud4qs38zl4hd03ifoyg2ksvfjkhuxebfzh3ef89vwvtvrr&ctrlRangeStart=0x0000000000000000&ctrlRangeEnd=0xffffffffffffffff&eventOffset=123&duration=1000
+```
 
-**Returns:**
+#### Parameters:
 
-- *range* <Array<string>> - a Recon message, representing the latest state of the range described by the subscription request. In the form of `[eventid, ahash, eventid]`
+* *streamid* <string> - the stream to subscribe to (Normally `*` is used to subscribe to sets of streams)
 
----
+* *{sort-key}* <string> - the *sort-key* and *sort-value* to subscribe to (e.g. `model` and `kjzl6hvfrbw6c82mkud4qs38zl4hd03ifoyg2ksvfjkhuxebfzh3ef89vwvtvrr` respectively)
+* *ctrlRangeStart* <string> - hex encoded string describing the start of the range of controllers to sync (defaults to `0x0000000000000000`)
+* *ctrlRangeEnd* <string> - hex encoded string describing the end of the range of controllers to sync (defaults to `0xffffffffffffffff`)
+* *eventOffset* <integer> - the number of events already consumed
+* *duration* <integer> - the amount of time, in seconds, to wait for a response
+
+#### Returns:
+
+There are two different content types supported as return value. 
 
+**JSON:**
 
+Content-Type:  `application/json`
 
-### `ceramic_unsubscribe`
+This content type returns a json representation of the events received. Note that this is just a representation of the event content, full verifiability is not retained.
 
-This method tells the Ceramic node to unsubscribe from a given interest range.
+- *events* <array<object>> - the list of events
+  - *eventId* <string> - the id of the event
+  - *id* <string> - CID of the InitEvent for this stream
+  - *prev* <array<string>> - CIDs of previous hash linked events
+  - *header* <object> - header for the event
+  - *data* <object> - the data of the event
+  - *timestamp* <integer> - the unixtime this event was timestamped (if it has been)
+- *eventOffset* <integer> - the number of events consumed
 
-**Params:**
 
-- *ceramicUrl* <string> - the Ceramic URL to unsubscribe from
-- *options* <object> - options for the subscription
-  - *ctrlRangeStart* <string> - hex encoded string describing the start of the range of controllers to sync (defaults to `0x0000000000000000`)
-  - *ctrlRangeEnd* <string> - hex encoded string describing the end of the range of controllers to sync (defaults to `0xffffffffffffffff`)
 
-**Returns:**
+**CAR file:**
```

Comment:
Need a line return  here, otherwise the indenting is broken in the rest of the file when viewed.

Reply:
That's a bug in the gitcoin renderer?

Reply:
Added more newlines, but that doesn't seem to fix the issue? Definitely seems like a bug.

Reply:
Yeah, maybe try a horizontal line to separate?

Reply:
That kind of breaks the general flow though. Anyway this won't be a problem on cips.ceramic.network